### PR TITLE
[CORE] Fix a potential issue in hive table scan when DPP is used

### PR DESCRIPTION
**Pull request by PHILO-HE**
The filter pruning expression of hive table scan may come from DPP. We need to do the check and transform for the pruning expression, the corresponding subqueryBroadcast and the possible re-used broadcast exchange. Otherwise, a mismatched subqueryBroadcast can be used and cause hash relation cast issues.

## How was this patch tested?
No regression issue in spark UTs.

### CODE REVIEW

**Review by rui-mo — COMMENTED** [review]
Can we desribe a bit in the PR description what does this PR trying to solve, and what is the potential issue?

**Comment by PHILO-HE** [comment]
> Can we desribe a bit in the PR description what is this PR trying to solve, and what is the potential issue?

Done.

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 1]
```diff
@@ -572,6 +574,20 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
         TransformHints.tagNotTransformable(newSource, validationResult.reason.get)
         newSource
       }
+    case plan if HiveTableScanExecTransformer.isHiveTableScan(plan) =>
+      // TODO: Add DynamicPartitionPruningHiveScanSuite.scala
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:578]: Is this suite disabled now?

**Review by rui-mo — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala (hunk 1 above)
Comment (PHILO-HE) [reply gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:578]: It's just introduced in spark 3.3. And may already be in our 3.3 UT plan list.

**Pull request merged**